gdal/port/cpl_port.h

```diff
@@ -338,7 +338,11 @@ typedef unsigned int  GUIntptr_t;
 
 #ifndef CPL_DLL
 #if defined(_MSC_VER) && !defined(CPL_DISABLE_DLL)
-#  define CPL_DLL     __declspec(dllexport)
+#  ifdef GDAL_EXPORTS
```

Comment:
This change the current behaviour of GDAL. Is it needed ? If such, then please revert the logic with for example #ifndef GDAL_IMPORTS

Reply:
Sorry, GDAL_EXPORTS is a define that is specific to our build environment.

However, I believe that the introduction of GDAL_IMPORTS would break application code.

Instead, I think that DLLBUILD, configured in nmake.opt should be used. Hence, in total:
   
    #ifndef CPL_DLL
    #if defined(_MSC_VER)
    #  ifdef DLLBUILD
    #    define CPL_DLL __declspec(dllexport)
    #  elif !defined(CPL_DISABLE_DLL)
    #    define CPL_DLL __declspec(dllimport)
    #  else
    #    define CPL_DLL
    #  endif
    #else
    #  if defined(USE_GCC_VISIBILITY_FLAG)
    #    define CPL_DLL     __attribute__ ((visibility("default")))
    #  else
    #    define CPL_DLL
    #  endif
    #endif
    #endif

Is that fine?

Reply:
Could you just revert that change ? I think for your purposes adding -DCPL_DLL= or -DCPL_DLL=__declspec(dllimport) to your nmake.opt should be enough